AzureAPIManagement/Services/ApiManagementClientBase.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Newtonsoft.Json;
+using SmallStepsLabs.Azure.ApiManagement.Utilities;
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SmallStepsLabs.Azure.ApiManagement
+{
+    public abstract class ApiManagementClientBase
+    {
+        private string Host { get; set; }
+
+
+        private string GetToken()
+        {
+            // Programmatically generate the access token used to call the API Management REST API.
+            // See "To programmatically create an access token" for instructions on how to get
+            // the values for id and key:
+            // https://msdn.microsoft.com/library/azure/5b13010a-d202-4af5-aabf-7ebc26800b3d#ProgrammaticallyCreateToken
+            // id - the value from the identifier text box in the credentials section of the
+            //      API Management REST API tab of the Security section.
+            string id = "<your identifier value here>";
```

Comment:
Needs to be parameterized and set on constructor. Environments will determine which keys are used (we have development and production for example).